src/nodes/FunctionCall.js

```diff
@@ -1,48 +1,61 @@
+/* eslint-disable implicit-arrow-linebreak */
 const {
   doc: {
     builders: { concat, group, indent, join, line, softline }
   }
 } = require('prettier');
 
-const FunctionCall = {
-  print: ({ node, path, print }) => {
-    let doc;
-    if (node.names && node.names.length > 0) {
-      doc = concat([
-        '{',
-        group(
-          concat([
-            indent(
-              concat([
-                softline,
-                join(
-                  concat([',', line]),
-                  path
-                    .map(print, 'arguments')
-                    .map((arg, index) => concat([node.names[index], ': ', arg]))
-                )
-              ])
-            ),
-            softline
-          ])
-        ),
-        '}'
-      ]);
-    } else {
-      doc = group(
+const printObject = (node, path, print) =>
+  group(
+    concat([
+      '{',
+      indent(
         concat([
-          indent(
-            concat([
-              softline,
-              join(concat([',', line]), path.map(print, 'arguments'))
-            ])
-          ),
-          softline
+          softline,
+          join(
+            concat([',', line]),
+            path
+              .map(print, 'arguments')
+              .map((arg, index) => concat([node.names[index], ': ', arg]))
+          )
         ])
-      );
-    }
-    return concat([path.call(print, 'expression'), '(', doc, ')']);
+      ),
+      softline,
+      '}'
+    ])
+  );
+
+const printParameters = (node, path, print) =>
+  group(
+    concat([
+      indent(
+        concat([
+          softline,
+          join(concat([',', line]), path.map(print, 'arguments'))
+        ])
+      ),
+      softline
+    ])
+  );
+
+const printArguments = (node, path, print) => {
+  if (node.names && node.names.length > 0) {
+    return printObject(node, path, print);
   }
+  if (node.arguments && node.arguments.length > 0) {
```

Comment:
This doesn't come from this PR, but I'm curious. When does a function call have "names" and when it has "arguments"?

Reply:
Will have to double check this one but from the look of the output code, a function call has names when is initializing a struct.
```Solidity
struct FooStruct {
  uint foo;
  uint figther;
}

FooStruct myStruct = FooStruct({foo:1, fighter:2});
```

Reply:
Oh, I see, thanks!